### PR TITLE
Add watchdog tutorial.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: docs/tutorials/misc/watchdog.mdx [hunk 1]
```diff
@@ -0,0 +1,175 @@
+# Watchdogs
+This tutorial will show you how to use the watchdogs to monitor the system and reset it if it hangs.
```
Comment (kasperl) [inline docs/tutorials/misc/watchdog.mdx:2]:
```suggestion
This tutorial will show you how to use watchdogs to monitor the system and reset it if it hangs.
```

File: docs/tutorials/misc/watchdog.mdx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+## Prerequisites
+We assume that you have set up your development environment as described
+in [the IDE tutorial](../../setup/ide).
+
+We also assume that you have flashed your device with Jaguar and that
+you are familiar with running Toit programs on it.
+If not, have a look at the [Hello world](../../setup/firstprogram) tutorial.
+
+Watchdogs use services. While not necessary, you may want to read the
+[Services](../../containers/services) tutorial to learn more about them.
```
Comment (kasperl) [inline docs/tutorials/misc/watchdog.mdx:18]:
```suggestion
[services](../../containers/services) tutorial to learn more about them.
```

File: docs/tutorials/misc/watchdog.mdx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  // Start the watchdog provider.
+  provider.main
+
+  // Create a watchdog client that connects to the provider.
+  client := WatchdogServiceClient
+  // Connect to the provider that has been started earlier.
+  client.open
+
+  // Create a watchdog.
+  dog := client.create "io.toit.docs/tutorial/my-dog"
```
Comment (kasperl) [inline docs/tutorials/misc/watchdog.mdx:51]: Why is this not docs.toit.io?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: docs/tutorials/misc/watchdog.mdx (hunk 3 above)
Comment (floitsch) [reply docs/tutorials/misc/watchdog.mdx]: got inspired (misled) by Java...
done.

**Pull request merged**